cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -90,6 +97,29 @@ func CreatePool(cStorPool *apis.CStorPool) error {
 	return nil
 }
 
+func validateDiskCount(pool *apis.CStorPool) error {
+	poolType := pool.Spec.PoolSpec.PoolType
+	diskCount := len(pool.Spec.Disks.DiskList)
+	diskCountEnv := getDiskCountEnv()
+	if diskCountEnv == 0 {
+		diskCountEnv = defaultDiskCount
+	}
+
+	if diskCount < defaultDiskCount {
+		return fmt.Errorf("Expected minimum %v no of disks, but got %v for pool type: %v", defaultDiskCount, diskCount, poolType)
+	}
+	return nil
+}
+
+func getDiskCountEnv() int {
```

Comment:
you should read this spc with default being '3' for raidz and '6' for raidz2 rather than env variable